File: PROGRAMMERS/두개뽑아서더하기.py
numbers = [2,1,3,4,1]
# numbers = [5,0,2,7]
result = []
num = 0
for i in range(num, len(numbers)):
    for j in range(num+1, len(numbers)):
        tmp = 0
        tmp = numbers[i] + numbers[j]
        result.append(tmp)
    num += 1

print(result)
result.sort()
print(set(result))

# def solution(numbers):
#     answer = []
#     for i in range(len(numbers)):
#         for j in range(i+1, len(numbers)):
#             answer.append(numbers[i] + numbers[j])
#     return sorted(list(set(answer)))

File: PROGRAMMERS/JAVA/src/Lv_1/두개뽑아서더하기.java
package Lv_1;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class 두개뽑아서더하기 {
    public int[] solution(int[] numbers) {
        List<Integer> l_nums = new ArrayList<>();
        for (int i = 0; i < numbers.length; i++) {
            for (int j = i + 1; j < numbers.length; j++) {
                l_nums.add(numbers[i] + numbers[j]);
            }
        }
        l_nums = l_nums.stream().distinct().collect(Collectors.toList());
        int[] answer = new int[l_nums.size()];
        Collections.sort(l_nums);
        for (int i = 0; i < l_nums.size(); i++) {
            answer[i] = l_nums.get(i).intValue();
        }
        return answer;
    }
}
